back/lib/citizen_lab/database/schema_postprocessor.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+class CitizenLab::Database::SchemaPostprocessor
+  attr_accessor :original_schema
+
+  def initialize(original_schema)
+    @original_schema = original_schema
+  end
+
+  def process
+    original_schema.dup.tap do |schema|
+      install_extensions!(schema)
+    end
+  end
+
+  private
+
+  # We have to install DB extensions as a post-processing step because there seems to be
+  # no way to configure Rails to dump the exact schema we need with the extensions
+  # included.
+  #
+  # By default, `rake db:schema:dump` does what we need and dumps only the schemas of
+  # the search path (as defined in `config/database.yml`). However, when dumping a
+  # specific set of schemas, the `pg_dump` call generated by `rake db:schemas:dump`
+  # includes the `--schema` flag. This means that no `CREATE EXTENSION` statements will
+  # be created in the resulting `structure.sql`. In other words, the extensions will be
+  # missing, causing any attempt to load the file to fail. See
+  # https://github.com/rails/rails/issues/17157#issuecomment-77390997) for more
+  # information.
+  #
+  # The alternative would be to set `config.active_record.dump_schemas = :all` in
```

Comment:
If we add a new extension, what should we do? It would be quite some task to edit `structure.sql` in this case.

Would it be possible to write this?
```ruby
config.active_record.dump_schemas = 'public,shared_extensions'
```
https://rubydoc.info/github/rails/rails/ActiveRecord:dump_schemas

Reply:
Unfortunately, we would still have to implement two things:
- a migration (for production/staging deployments),
- modify the schema post-processor to add the new extension.

But that's not much different from the previous situation where, in addition to implementing a new migration, we had to adapt some rake tasks that this PR removes. All in all, I think that's okay, since it's not something we do very often and even if it's bothersome, it's not that complex to implement.

```
config.active_record.dump_schemas = 'public,shared_extensions'
```
would produce the exact same `structure.sql` (i.e., without the extensions) and would still require a post-processing step.